packages/vite/src/node/config.ts

```diff
@@ -616,118 +620,6 @@ function resolveBaseUrl(
   return base
 }
 
-function mergeConfigRecursively(
```

Comment:
This is moved to the utils file. Is the reason because `config.ts` contains side-effects that pollutes the final bundle. Maybe it's worth investigating and prevent it 🤔

Reply:
Yes. There are quite some side effects.